lms/djangoapps/courseware/courses.py

```diff
@@ -572,6 +572,9 @@ def get_course_assignments(course_key, user, include_access=False):  # lint-amne
                     url = reverse('jump_to', args=[course_key, subsection_key])
 
                 complete = is_block_structure_complete_for_assignments(block_data, subsection_key)
+                if not assignment_released:
+                    complete = False
```

Comment:
There are no desired side effects of `is_block_structure_complete_for_assignments` right? If not, I'd vote for skipping it with something like:

```
if assignment_released:
    complete = is_block_structure_complete_for_assignments(block_data, subsection_key)
else:
    complete = False
```

Or fancier (whether it's clearer probably depends on how much you live and breathe Python):
```
complete = assignment_released and is_block_structure_complete_for_assignments(block_data, subsection_key)
```